### PR TITLE
Feature request #96

### PR DESCRIPTION
**Pull request by CristianSarac**
Added a background image on card selected ,just like discused  in #96.

There is also a small fix for the card images since the images were cut or cropped  

I havent added this option on the Home screen since that one is pretty crowded with the large  card at the top. 
Does it make sense to add  this option there too  ?

### CODE REVIEW

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt [hunk 1]
```diff
@@ -49,6 +69,46 @@ class SeasonBrowseFragment : BrowseSupportFragment(), OnItemViewClickedListener
...
+    /**
+     * Function that will load a [Session.largePictureUrl] as a background image when a card is selected
+     */
+    private fun initializeBackground() {
+        val backgroundManager = BackgroundManager.getInstance(activity).apply {
+            attach(activity?.window)
+        }
+
+        val metrics = DisplayMetrics()
+        requireActivity().windowManager.defaultDisplay.getMetrics(metrics)
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt:81]: Hm, this could actually mean that you have a 1080p image loaded in a custom target of 720p (in some TVs), which, might be fine but how does it looks?

File: app/src/main/java/fr/groggy/racecontrol/tv/core/season/SeasonService.kt [hunk 2]
```diff
@@ -99,7 +100,7 @@ class SeasonService @Inject constructor(
 
     private fun thumbnail(session: F1TvSession): Flow<Image> {
         return flowOf(
-            Image(Uri.parse(session.pictureUrl))
+            Image(Uri.parse(session.largePictureUrl)) // We use the large picture url since a better quality image is preferred on a large screen
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/core/season/SeasonService.kt:103]: Revert this change.

The thing is that doing this we are caching more stuff, for some devices as the Google Chrome cast with google tv, we can easily fill up the storage, and also the card isn't 1080p, it is just a waste.

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt [hunk 3]
```diff
@@ -49,6 +69,46 @@ class SeasonBrowseFragment : BrowseSupportFragment(), OnItemViewClickedListener
...
+
+        onItemViewSelectedListener = OnItemViewSelectedListener { _, item, _, _ ->
+            if (item is Session) {
+                viewLifecycleOwner.lifecycleScope.launch {
+                    delay(BACKGROUND_UPDATE_DELAY) // A small delay so that the loading of the image and transition is more smooth
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt:86]: can you elaborate this comment a bit better?

Why is it more smooth ?

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt [hunk 4]
```diff
@@ -49,6 +69,46 @@ class SeasonBrowseFragment : BrowseSupportFragment(), OnItemViewClickedListener
...
+
+                    Glide.with(requireActivity())
+                        .load(item.largePictureUrl)
+                        .centerCrop()
+                        .error(R.drawable.banner) // In case of error we show the banner , maybe a error image could be better ?
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt:91]: This comment is not required

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/session/SessionCardPresenter.kt [hunk 5]
```diff
@@ -16,9 +16,10 @@ class SessionCardPresenter @Inject constructor(
     private val settingsRepository: SettingsRepository
 ) : Presenter() {
 
+    // We set the size of the Card to be 1/4 of the 1920 x 1080 since we use the LargePictureUrl , this way we preserve aspect ratio and images wont be cut or cropped
     companion object {
-        private const val WIDTH = 313
-        private const val HEIGHT = 176
+        private const val WIDTH = 480
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/session/SessionCardPresenter.kt:21]: For this I think is better to then change the original image width / height (the url) to this specific size, so it won't crop as well.

**Review by leonardoxh — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt (hunk 3 above)
Comment (CristianSarac) [reply app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt:86]: This will make sure the previous image has finished loading (if that is the case) and the new image will start loading a bit later.I hope it make sense

File: app/src/main/java/fr/groggy/racecontrol/tv/core/season/SeasonService.kt [hunk 6]
```diff
@@ -88,6 +88,7 @@ class SeasonService @Inject constructor(
                         id = session.id,
                         contentId = session.contentId,
                         name = session.name,
+                        largePictureUrl =session.largePictureUrl,
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/core/season/SeasonService.kt:91]: Whitespace after `=`

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseViewModel.kt [hunk 7]
```diff
@@ -50,6 +50,7 @@ data class Session(
     override val name: String,
     override val contentSubtype: String,
     override val thumbnail: Image?,
+    val largePictureUrl:String,
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseViewModel.kt:53]: whitespace  `:`

**Review by leonardoxh — COMMENTED** [review]
Looks good, I will perform testing after these last 2 comments.

Would you mind to add an updated screenshot of the feature?

File: app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt [hunk 8]
```diff
@@ -49,6 +69,46 @@ class SeasonBrowseFragment : BrowseSupportFragment(), OnItemViewClickedListener
...
+        requireActivity().windowManager.defaultDisplay.getMetrics(metrics)
+
+        onItemViewSelectedListener = OnItemViewSelectedListener { _, item, _, _ ->
+            if (item is Session) {
+                viewLifecycleOwner.lifecycleScope.launch {
+                    delay(BACKGROUND_UPDATE_DELAY) // A small delay so that the loading of the image and transition is more smooth
+
+                    Glide.with(requireActivity())
+                        .load(item.largePictureUrl)
+                        .centerCrop()
```
Comment (leonardoxh) [inline app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt:90]: Hm, can you test this in a 720p tv emulator or so?

What does this centerCrop does there?

Is it really needed?

Comment (CristianSarac) [reply app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt:90]: I can no longer test it since my account has ended and i wont be renewing it untill the 2022 season starts.
And as for screenshoots the most recent ones that i have are in here -~>  #37

Comment (leonardoxh) [reply app/src/main/java/fr/groggy/racecontrol/tv/ui/season/browse/SeasonBrowseFragment.kt:90]: :D Same for me, I am merging it then will try out.
